packages/resilient/attachments/init.lua

```diff
@@ -0,0 +1,358 @@
+    -- So we just create a simple uncompressed stream here, and the library will do the rest for us.
+    local streamStr = table.concat({
+      "<<",
+        "/Type /EmbeddedFile",
+        "/Subtype /" .. mime:gsub("/", "#2F"),
+        "/Length",
+        string.len(data),
+      ">>\nstream\n" .. data .. "\nendstream"
+    }, " ")
+
+    local stream = pdf.parse(streamStr)
+    local streamRef1 = pdf.reference(stream) -- for /F entry in /EF dictionary
+    local streamRef2 = pdf.reference(stream) -- for /UF entry in /EF dictionary
+    pdf.release(stream)
+
+    -- Create the file specification dictionary
+    -- <<
+    --   /Type /Filespec
+    --   /F (...)                % File name (ASCII mandatory, so we'll use placeholder name to avoid issues)
+    --   /UF <...>               % Unicode file name (UTF-16BE hex-encoded), optional but we will always set it
+    --   /Desc <...>             % Description (optional)
+    --   /EF <<
+    --         /F REF            % Object reference for the embedded file stream with ASCII name
+    --         /UF REF           % Ooptional reference for the embedded file stream with Unicode name,
+    --                           %            Recommended to be present and identical to /F entry
+    --       >>
+    --   /AFRelationship /Data   % Relationship type (Data, Source, Supplement, Alternative, Unspecified)
+    -- >>
+    local fileStr = table.concat({
+      "<<",
+        "/Type /Filespec",
+        "/F (attachment" .. n .. ".txt)",
+        "/UF <" .. u16name .. ">",
+        "/Desc <" .. u16desc .. ">",
+        "/AFRelationship /" .. (entry.relation or "Data"),
+      ">>"
+    }, " ")
+    local fileSpec = pdf.parse(fileStr)
+    local efDict = pdf.parse("<< >>")
+    pdf.add_dict(efDict, pdf.parse("/F"), streamRef1)
+    pdf.add_dict(efDict, pdf.parse("/UF"), streamRef2)
+    pdf.add_dict(fileSpec, pdf.parse("/EF"), efDict)
+
+    local fileSpecRef1 = pdf.reference(fileSpec) -- for /AF array in Catalog
+    local fileSpecRef2 = pdf.reference(fileSpec) -- for /Names array in /EmbeddedFiles in Catalog
+    pdf.release(fileSpec)
+
+    SU.debug("resilient.attachments", "Built attachment PDF objects for", name)
+
+    attachments:update({
+      [u16name] = {
+        fileSpecRef1 = fileSpecRef1,
+        fileSpecRef2 = fileSpecRef2
+      }
+    })
+  end
+  return attachments
+end
+
+--- (Private) Output attachments to the PDF.
+--
+-- Called at the end of the document generation process.
+function package:_outputAttachments ()
+  SILE.outputter:_ensureInit()
+  pdf = require("justenoughlibtexpdf")
+
+  local attachments = self:_buildAttachments()
+  local n = #attachments:keys()
+  if n == 0 then
+    SU.debug("resilient.attachments", "No attachments to output")
+    return -- Nothing to do
+  end
+  SU.debug("resilient.attachments", "Outputting", n, "attachments to PDF")
+
+  -- Build an /AF array (PDF/A-3u requirement, used by some tools and mandatory for invoices in Factur-X).
```

Comment:
PDF A-3, not A-3u.

We need to make this optional. While mandated by Factur-X for invoices, not all embedded files have to be here.
 - /Names/EmbeddedFiles = All embedded files
 - /AF = Associated Files, considered as integral part of the document, i.e. usually a subset of the former.